core/templates/dev/head/domain/statistics/PlaythroughObjectFactory.js

```diff
@@ -0,0 +1,73 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating new frontend instances of Playthrough
+ *     domain objects.
+ */
+
+oppia.factory('PlaythroughObjectFactory', [
+  'LearnerActionObjectFactory', function(LearnerActionObjectFactory) {
+    /**
+     * @constructor
+     * @param {string} playthroughId - ID of a playthrough.
+     * @param {string} expId - ID of an exploration.
+     * @param {number} expVersion - Version of an exploration.
+     * @param {string} issueType - type of an issue.
+     * @param {*} issueCustomizationArgs - customization dict for an issue.
+     * @param {list(LearnerAction)} actions - list of learner actions.
+     */
+    var Playthrough = function(playthroughId, expId, expVersion, issueType,
+        issueCustomizationArgs, actions) {
+      this.playthroughId = playthroughId;
+      this.expId = expId;
+      this.expVersion = expVersion;
+      this.issueType = issueType;
+      this.issueCustomizationArgs = issueCustomizationArgs;
+      this.actions = actions;
+    };
+
+    /**
+     * @param {string} playthroughId - ID of a playthrough.
+     * @param {string} expId - ID of an exploration.
+     * @param {number} expVersion - Version of an exploration.
+     * @param {string} issueType - type of an issue.
+     * @param {*} issueCustomizationArgs - customization dict for an issue.
+     * @param {list(LearnerAction)} actions - list of learner actions.
+     * @returns {Playthrough}
+     */
+    Playthrough.create = function(
+        playthroughId, expId, expVersion, issueType, issueCustomizationArgs,
+        actions) {
+      return new Playthrough(
+        playthroughId, expId, expVersion, issueType, issueCustomizationArgs,
+        actions);
+    };
+
+    /**
+     * @param {{playthroughId: string, expId: string, expVersion: number, issueType: string, issueCustomizationArgs, actions: list(LearnerAction)}} playthroughBackendDict
```

Comment:
@brianrodri Not sure how to fit this into one line. Think you could help?

Reply:
Same as above.

Reply:
Done.